core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -54,9 +66,35 @@
   private Executor executor;
   private final List<ClientInterceptor> interceptors = new ArrayList<ClientInterceptor>();
 
+  private final URI target;
+
+  @Nullable
+  private final SocketAddress directServerAddress;
+
   @Nullable
   private String userAgent;
 
+  @Nullable
+  private NameResolver nameResolver;
+
+  @Nullable
+  private LoadBalancer loadBalancer;
+
+  protected AbstractManagedChannelImplBuilder(URI target) {
+    this.target = Preconditions.checkNotNull(target);
+    this.directServerAddress = null;
+  }
+
+  protected AbstractManagedChannelImplBuilder(SocketAddress directServerAddress) {
```

Comment:
This doesn't look like a great idea to me. I'd much rather Netty handle this, especially since most of the SocketAddresses we expect to be passed to Netty are Netty-specific.